flexmeasures/data/models/time_series.py

```diff
@@ -349,8 +362,61 @@ def search(
             source=parsed_sources,
             most_recent_beliefs_only=most_recent_beliefs_only,
             most_recent_events_only=most_recent_events_only,
+            custom_filter_criteria=source_criteria,
         )
 
+    def collect(
```

Comment:
1. Isn't this a function we only use for the data model migration transition period? At the very least there should be a comment to that effect, so we find it later when we need to deprecate.
2. I might also need an explanation why we have a "collect" function both as class method and as instance method. It is quite confusing. This seems to call `search`, which *is* a class method...

Reply:
Thanks for noticing: the new `collect` function is now a class method, too (of `TimedBelief`). The old `collect` function is a class method of `TimedValue`.

We also discussed renaming things by prepending `legacy_`. I think we should rename the `TimedValue` class rather than it's `collect` method, and I propose to rename legacy code in a separate PR, perhaps as the last PR within project 9?

Reply:
Renaming later is a sound idea. In my work, I'm replacing code, and I probably don't want to change the name of modules only to avoid a clash (e.g. for "dashboard.py"). But I'll look into it.